stdlib/public/core/VarArgs.swift

```diff
@@ -467,6 +491,140 @@ final internal class _VaListBuilder {
   internal var storage: ContiguousArray<Int>
 }
 
+#elseif arch(arm64) && !(os(macOS) || os(iOS) || os(tvOS) || os(watchOS))
+
+
+@_fixed_layout // FIXME(sil-serialize-all)
+@_versioned // FIXME(sil-serialize-all)
+final internal class _VaListBuilder {
+
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned // FIXME(sil-serialize-all)
+  internal init() {
+    // prepare the register save area
+    allocated = _registerSaveWords
+    storage = allocStorage(wordCount: allocated)
+    // append stack arguments after register save area
+    count = allocated
+  }
+
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned // FIXME(sil-serialize-all)
+  deinit {
+    if let allocatedStorage = storage {
+      deallocStorage(wordCount: allocated, storage: allocatedStorage)
+    }
+  }
+
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned // FIXME(sil-serialize-all)
+  internal func append(_ arg: CVarArg) {
+    var encoded = arg._cVarArgEncoding
+
+    if arg is _CVarArgPassedAsDouble
+      && fpRegistersUsed < _countFPRegisters {
+        var startIndex = (fpRegistersUsed * _fpRegisterWords)
+          for w in encoded {
+            storage[startIndex] = w
+            startIndex += 1
+          }
+          fpRegistersUsed += 1
+    }else if encoded.count == 1
+        && !(arg is _CVarArgPassedAsDouble)
+        && gpRegistersUsed < _countGPRegisters {
+        var startIndex = ( _fpRegisterWords * _countFPRegisters) + gpRegistersUsed
+        storage[startIndex] = encoded[0]
+        gpRegistersUsed += 1
+    }else{
+        //arguments in stack slot
+        appendWords(encoded)
+    }
+  }
+
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned // FIXME(sil-serialize-all)
+  internal func va_list() -> CVaListPointer {
+    let vr_top = storage + (_fpRegisterWords * _countFPRegisters)
+    let gr_top = vr_top + _countGPRegisters
+
+    return CVaListPointer(__stack: gr_top, __gr_top: gr_top,
+                          __vr_top: vr_top, __gr_off: -64, __vr_off: -128)
+  }
+
+  @_inlineable // FIXME(sil-serialize-all)
+  @_versioned // FIXME(sil-serialize-all)
+  internal func appendWords(_ words: [Int]) {
+    let newCount = count + words.count
+    if newCount > allocated {
+      let oldAllocated = allocated
+      let oldStorage = storage
+      let oldCount = count
+
+      allocated = max(newCount, allocated * 2)
+      let newStorage = allocStorage(wordCount: allocated)
+      storage = newStorage
+      // count is updated below
```

Comment:
Please write comments in sentence case, with a beginning capital letter and a closing period.